nemo/collections/nlp/models/dialogue/dialogue_gpt_generation_model.py

```diff
@@ -275,10 +276,11 @@ def generate_candidates(self, labels, template_length, input_ids, attn_masks):
             for i in range(input_ids.size(0)):
                 param_dict = {
                     "input_ids": input_ids[i : i + 1, : template_length[i]],
-                    "attention_masks": attn_masks[i : i + 1, : template_length[i]],
                     "max_length": template_length[i] + tokens_to_generate,
                     "pad_token_id": self.tokenizer.tokenizer.pad_token_id,
                 }
+                if "attention_mask" in set(inspect.signature(self.language_model.generate).parameters):
+                    param_dict["attention_masks"] = attn_masks[i: i + 1, : template_length[i]]
```

Comment:
I'm unsure if this fix is correct, however after [this](https://github.com/huggingface/transformers/commit/ed1924e801c4d7e484703dc4624c185b8da5d5a2) commit in transformers, I think attention_masks was silently not being used